kubectl-plugin/pkg/cmd/get/get_nodes.go

```diff
@@ -0,0 +1,265 @@
+	cmdFactory := cmdutil.NewFactory(options.configFlags)
+
+	cmd := &cobra.Command{
+		Use:          "node [NODE] [(-c|--ray-cluster) RAYCLUSTER]",
+		Aliases:      []string{"nodes"},
+		Short:        "Get Ray nodes",
+		Example:      getNodesExample,
+		SilenceUsage: true,
+		Args:         cobra.MaximumNArgs(1),
+		// ValidArgsFunction: TODO
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := options.Complete(args); err != nil {
+				return err
+			}
+			if err := options.Validate(); err != nil {
+				return err
+			}
+			k8sClient, err := client.NewClient(cmdFactory)
+			if err != nil {
+				return fmt.Errorf("failed to create client: %w", err)
+			}
+			return options.Run(cmd.Context(), k8sClient)
+		},
+	}
+
+	cmd.Flags().StringVarP(&options.cluster, "ray-cluster", "c", "", "Ray cluster")
+	cmd.Flags().BoolVarP(&options.allNamespaces, "all-namespaces", "A", false, "If present, list nodes across all namespaces. Namespace in current context is ignored even if specified with --namespace.")
+
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *GetNodesOptions) Complete(args []string) error {
+	if options.allNamespaces {
+		options.namespace = ""
+	} else {
+		if options.configFlags.Namespace == nil || *options.configFlags.Namespace == "" {
+			options.namespace = "default"
+		} else {
+			options.namespace = *options.configFlags.Namespace
+		}
+	}
+
+	if len(args) > 0 {
+		options.node = args[0]
+	}
+
+	return nil
+}
+
+func (options *GetNodesOptions) Validate() error {
+	// Overrides and binds the kube config then retrieves the merged result
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("error retrieving raw config: %w", err)
+	}
+	if !util.HasKubectlContext(config, options.configFlags) {
+		return fmt.Errorf("no context is currently set, use %q or %q to select a new one", "--context", "kubectl config use-context <context>")
+	}
+	return nil
+}
+
+func (options *GetNodesOptions) Run(ctx context.Context, k8sClient client.Client) error {
+	listopts := v1.ListOptions{
+		LabelSelector: joinLabelMap(createLabelSelectors(options.cluster)),
+	}
+
+	if options.node != "" {
+		listopts.FieldSelector = fmt.Sprintf("metadata.name=%s", options.node)
+	}
+
+	pods, err := k8sClient.KubernetesClient().CoreV1().Pods(options.namespace).List(ctx, listopts)
+	if err != nil {
+		return fmt.Errorf("unable to get Ray nodes: %w", err)
+	}
+
+	nodes := podsToNodes(pods.Items)
+
+	if options.node != "" && len(nodes) == 0 {
+		return errors.New(errorMessageForNodeNotFound(options.node, options.cluster, options.namespace, options.allNamespaces))
+	}
+
+	return printNodes(nodes, options.allNamespaces, options.ioStreams.Out)
+}
+
+// createLabelSelectors creates a map of K8s label selectors for Ray nodes
+func createLabelSelectors(cluster string) map[string]string {
+	labelSelectors := map[string]string{
+		util.RayNodeTypeLabelKey: string(rayv1.WorkerNode),
```

Comment:
Thoughts on listing head pods too and adding a column for node type (HEAD or WORKER)?

Reply:
ah yes, sounds good